lms/djangoapps/commerce/api/v1/tests/test_views.py

```diff
@@ -287,25 +287,45 @@ def test_remove_upgrade_deadline(self):
         self.assertIsNone(updated_verified_mode.expiration_datetime)
 
     def test_update_overwrite(self):
-        """ Verify that data submitted via PUT overwrites/deletes modes that are
-        not included in the body of the request. """
-        course_id = six.text_type(self.course.id)
-        expected_course_mode = CourseMode(
+        """
+        Verify that data submitted via PUT overwrites/deletes modes that are
+        not included in the body of the request, EXCEPT the Masters mode,
+        which it leaves alone.
+        """
+        existing_mode = self.course_mode
+        existing_masters_mode = CourseMode.objects.create(
+            course_id=self.course.id,
+            mode_slug=u'masters',
+            min_price=10000,
+            currency=u'USD',
+            sku=u'DEF456',
+            bulk_sku=u'BULK-DEF456'
+        )
+        new_mode = CourseMode(
+            course_id=self.course.id,
             mode_slug=u'credit',
             min_price=500,
             currency=u'USD',
             sku=u'ABC123',
             bulk_sku=u'BULK-ABC123'
         )
-        expected = self._serialize_course(self.course, [expected_course_mode])
-        path = reverse('commerce_api:v1:courses:retrieve_update', args=[course_id])
-        response = self.client.put(path, json.dumps(expected), content_type=JSON_CONTENT_TYPE)
+
+        path = reverse('commerce_api:v1:courses:retrieve_update', args=[six.text_type(self.course.id)])
+        data = json.dumps(self._serialize_course(self.course, [new_mode]))
+        response = self.client.put(path, data, content_type=JSON_CONTENT_TYPE)
         self.assertEqual(response.status_code, 200)
-        actual = json.loads(response.content)
-        self.assertEqual(actual, expected)
 
-        # The existing CourseMode should have been removed.
-        self.assertFalse(CourseMode.objects.filter(id=self.course_mode.id).exists())
+        # Check modes list in response, disregarding its order.
+        expected_dict = self._serialize_course(self.course, [new_mode])
+        expected_items = expected_dict['modes']
+        actual_items = json.loads(response.content)['modes']
+        self.assertCountEqual(actual_items, expected_items)
+
+        # The existing non-Masters CourseMode should have been removed.
```

Comment:
I appreciate all the comments 👍 looks good to me

Reply:
Great!  I didn't write them! :P